packages/blocks/src/multiple-choice-question/index.js

```diff
@@ -4,21 +4,36 @@
 import { RichText } from '@wordpress/block-editor';
 import { createContext } from '@wordpress/element';
 import classnames from 'classnames';
+import { isEmpty } from 'lodash';
+import { __ } from '@wordpress/i18n';
 
 /**
  * Internal dependencies
  */
-import { QuestionHeader, QuestionWrapper } from '../components';
+import { ErrorMessage, QuestionHeader, QuestionWrapper } from '../components';
 import { Style } from './constants';
+import { useValidation } from '@crowdsignal/form';
 
 const Context = createContext();
 
 const MultipleChoiceQuestion = ( { attributes, children, className } ) => {
+	const { error } = useValidation( {
+		fieldName: `q_${ attributes.clientId }[choice]${
```

Comment:
Hmm, the idea of `useValidation` was it'd just validate the entire form state, so the callback would become something like:

```
useValidation( ( form ) => {
    if ( attributes.mandatory && isEmpty( form[ `q_${ attributes.clientId }[choice]` ] ) ) {
        ...
```

This will also enable us to enforce more complicated rules down the line - like, a question or a form field might only be mandatory when another option is selected.  
So that would make `useValidation` somewhat more universal, while `useField`'s `validation` prop should still only focus on the fields' value.

On another note, I'd be really nice to refactor the form system a little so things like multiple choice fields are actually arrays and not just a comma separated string. One of the reasons being validation - but it doesn't have to be done in this PR necessarily.  
Other reason being, survey submission relies on existing logic but I expect any new endpoints for anything else we do to use proper JSON, in which case this would become somewhat problematic maybe.

Reply:
Some time later (:spongebob)... I finally find some time to go through this.
Could you elaborate a bit more on how you imagined the `useValidation` hook?
From what I understood, it would receive a function as a parameter and, internally, get the form state and call the function passing the form state.
What I didn't understand well is how we'd set the error state in case one of the validations fail. Would we bring this logic to the callback function?

Reply:
Right, so the main difference you're now getting the entire `form` as the argument allowing you to validate any fields from any place within the form.  

As for how you'd set the error state, that callback should return an object with question names as keys and error messages as values.
If a key doesn't exist - that field is valid. Eventually we could maybe even support an array of error messages as values for some more complicated stuff but I'd say that's getting ahead of ourselves a little. For now, if you have multiple callbacks validating the same field, the errors would just override each other. Which is fine, you see one, fix it, then you see the other if it still applies.

Reply:
I like the idea. Wondering if this *field name* would need to be the same value as in `useField` where the value/name is an encoded name or if we should use a *human readable* or uniquely identifiable name (clientID?)

Given this question, how likely is it that we validate the form values depending on other field values? How would inputs know of each other *by name*?

Reply:
After discussing a few other points with @ice9js, we decide to keep the validation as-is for now and any necessary changes will be handled next.